File: SeriesTracker.Core/Models/Shikimori/Genre.cs
using Newtonsoft.Json;

namespace SeriesTracker.Core.Models.Shikimori
{
    /// <summary>
    /// Представляет информацию о жанре аниме.
    /// </summary>
    public class Genre
    {
        /// <summary>
        /// Уникальный идентификатор жанра.
        /// </summary>
        [JsonProperty("id")]
        public long Id { get; set; }

        /// <summary>
        /// Название жанра на русском языке.
        /// </summary>
        [JsonProperty("russian")]
        public required string Russian { get; set; }

        /// <summary>
        /// Тип жанра (например, "genre", "theme").
        /// </summary>
        [JsonProperty("kind")]
        public required string Kind { get; set; }
    }
}

File: client/app/Models/Genre.ts
/**
 * @interface Genre
 * @description Объект, представляющий жанр аниме.
 * @property {string} id Уникальный идентификатор жанра.
 * @property {string} russian Название жанра на русском языке.
 */
interface Genre {
    id: string;
    russian: string;
}
/**
 * @interface Genre
 * @description Объект, представляющий категоризированные жанры.
 * @property {Genre[]} genre Массивов жанров (фэнтези, романтика...).
 * @property {Genre[]} theme Массивов жанров 'тема' (средневековье, взрослые персонажи...).
 * @property {Genre[]} demographic Массивов жанров 'аудитория' (сенен, сэйнен...).
 */
interface GroupGenre {
    genre: Genre[];
    theme: Genre[];
    demographic: Genre[];
}
